### PR TITLE
Add User API docs

### PR DESCRIPTION
**Pull request by BarVolunteering**

### CODE REVIEW

**Comment by BarVolunteering** [comment]
I am unable to find @Danielsh28 in Github so i notified him by Slack.

File: docs/USERAPI.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
+# User API documentation
+
+This is the documentation for the new user API (and only the new one) that was created in 2020.
```
Comment (yuvalbl) [inline docs/USERAPI.md:3]: Not sure what you mean by that.
Of this API is for new users only - please clarify
If this is User API v2, please use the term version (Today's "New" will get old someday :) )

File: docs/USERAPI.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+### Method
+
+> GET
+
+### Example
+
+> https://www.anyway.co.il/authorize/google
+
+> http://127.0.0.1:5000/authorize/google?redirect_url=http%3A%2F%2F127.0.0.1%3A5000%2F
```
Comment (yuvalbl) [inline docs/USERAPI.md:26]: These are local addresses, is this a mistake?

File: docs/USERAPI.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+## Authorization
+
+### Description
+
+User authorization is been done with OAuth 2.0 and Google as the OAuth 2.0 provider. When calling the authentication URL you will
+be redirected to Google OAuth system (So the user need to be able to see the screen), after Google will authenticate the user he will be redirected
+to https://www.anyway.co.il/authorize/google/callback/google and from there the flow will be redirected to the URL that was given in
+the param `redirect_url` or to the default URL.
+
+### URL struct
```
Comment (yuvalbl) [inline docs/USERAPI.md:14]: It is difficult to read such documentation for multiple calls - each request has Struct, method, etc.

I suggest using smaller titles, and less of them:
```
### Authorization (GET | https://www.anyway.co.il/authorize/google)
User authorization is been...
##### Examples:
https://www.anyway.co.il/authorize/google
##### Returns:
...

```

Comment (yuvalbl) [reply docs/USERAPI.md:14]: However, I didn't see the MD parse result, only this PR, so you can ignore this comment if you like :)

**Review by yuvalbl — CHANGES_REQUESTED** [review]
Please see the first comment - this is the major one.

File: docs/USERAPI.md (hunk 1 above)
Comment (BarVolunteering) [reply docs/USERAPI.md:3]: Yes, we can call it User API v2,
Somewhere between 2012 to 2014 someone wrote a User API system that is still present in the BE code and is also present in production, this code is currently broken.
To remove it we to do (a big?) refactor as this system is integrated into other parts of the server, and as there are future plans to rewrite all of BE server code - this task has currently been postponed. 

I will update the docs accordingly.

File: docs/USERAPI.md (hunk 2 above)
Comment (BarVolunteering) [reply docs/USERAPI.md:26]: No, this is just to indicate that you perform local debugging of this feature(redirect your flow to localhost instead to the Anyway servers).

Comment (BarVolunteering) [reply docs/USERAPI.md:26]: The third example is showing how you can perform local to Anyway servers flow.

**Comment by atalyaalon** [comment]
@BarVolunteering can I merge?

**Comment by BarVolunteering** [comment]
> @BarVolunteering can I merge?

Yes

**Pull request merged**